docs/java/java-testing.md

```diff
@@ -21,39 +21,110 @@ The extension supports the following test frameworks:
+</dependency>
+```
+
+#### Gradle
+Make sure following lines are added in your `build.gradle`:
+
+```groovy
+plugins {
+    java
+}
+
+dependencies {
+    testImplementation('junit:junit:(YOUR_JUNIT_VERSION)')
+}
+```
+
+#### Unmanaged Folder
+If your project does not use any build tools. You need to download the following JARs and add them to the project classpath (via setting `java.project.referencedLibraries`):
+
+- [junit.jar](https://search.maven.org/search?q=g:junit%20AND%20a:junit)
+- [hamcrest-core.jar](https://search.maven.org/artifact/org.hamcrest/hamcrest-core/1.3/jar)
+
+> You can check the [official wiki](https://github.com/junit-team/junit4/wiki/Download-and-Install) for more information about how to setup JUnit 4.
+
+### JUnit 5
+The JUnit 5 team provides a collection of sample projects with different build tools. Check the [junit5-sample repository](https://github.com/junit-team/junit5-samples) if your project uses Maven or Gradle as your build tool.
+
+#### Unmanaged Folder
+If your project does not use any build tools, for simplicity, you can directly include the [junit-platform-console-standalone](https://repo1.maven.org/maven2/org/junit/platform/junit-platform-console-standalone/) JAR in the project classpath (via setting `java.project.referencedLibraries`).
+
+### TestNG
+#### Maven
+Add following configuration into your `pom.xml`:
+
+```xml
+<dependency>
+  <groupId>org.testng</groupId>
+  <artifactId>testng</artifactId>
+  <version>(YOUR_TESTNG_VERSION)</version>
+  <scope>test</scope>
+</dependency>
+```
+
+#### Gradle
+Make sure following lines are added in your `build.gradle`:
+
+```groovy
+plugins {
+    java
+}
+
+dependencies {
+    testImplementation('org.testng:testng:(YOUR_TESTNG_VERSION)')
+}
+```
+
+#### Unmanaged Folder
+If your project does not use any build tools. You need to download the following JARs and add them to the project classpath (via setting `java.project.referencedLibraries`):
+
+- [testng.jar](https://search.maven.org/search?q=g:org.testng%20AND%20a:testng)
+- [jcommander.jar](https://search.maven.org/artifact/com.beust/jcommander/1.81/jar)
+
 ## Features
 
 ### Run/Debug Test Cases
 
-The extension will generate shortcuts (the green play button) on the left side of the class and method definition. To run the target test cases, select the green play button. You can also right-click on the play button to see more options.
+The extension will generate shortcuts (the green play button) on the left side of the class and method definition. To run the target test cases, click the green play button. You can also right-click on it to see more options.
 
 ![Run/Debug Test Cases](images/java-testing/editor-decoration.png)
 
 ### Test Explorer
 
-The Test Explorer shows all the test cases in your workspace. You can also run/debug your test cases from there.
+The Test Explorer shows all the test cases in your workspace. You can also run/debug your test cases and view their test results from there.
```

Comment:
Let's assume people are not familiar with Test Explorer. So to add some background info

Something like:

Test Explorer is one of the main buttons on the left-side toolbar of Visual Studio Code. Once you click on this button, it will show all Java test cases in your workspace

Reply:
Ok